intera_interface/src/motion_interface/interaction_options.py

```diff
@@ -0,0 +1,243 @@
+        @param max_impedance (impedance modulation state):
+        --> None:  populate with vector of default values (True)
+        --> bool:  set each dimension by the input boolean
+        --> [bool]:  copy all elements. Checks length.
+        """
+        if max_impedance is None:
+            self._data.max_impedance =  [self.default_max_impedance]*self.n_dim_cart
+        elif len(max_impedance) == 1:
+            self._data.K_nullspace = [K_nullspace[0]]*self.n_dim_cart
+        elif len(max_impedance) == self.n_dim_cart:
+            self._data.max_impedance = deepcopy(max_impedance)
+        else:
+            rospy.logerr('The number of max_impedance must be 1 or %d',
+                         self.n_dim_cart)
+
+    def set_K_nullspace(self, K_nullspace = None):
+        """
+        @param K_nullspace (nullspace stiffness gains):
+        --> None:  populate with vector of default values
+        --> float:  set each dimension by the input float number
+        --> [float]:  copy all elements. Checks length.
+        """
+        if K_nullspace is None:
+            self._data.K_nullspace = [self.default_K_nullspace]*self.n_dim_joint
+        elif len(K_nullspace) == 1:
+            self._data.K_nullspace = [K_nullspace[0]]*self.n_dim_joint
+        elif len(K_nullspace) == self.n_dim_joint:
+            self._data.K_nullspace = deepcopy(K_nullspace)
+        else:
+            rospy.logerr('The number of K_nullspace must be 1 or %d',
+                         self.n_dim_joint)
+
+    def set_force_command(self, force_cmd = None):
+        """
+        @param force_command:
+        --> None:  populate with vector of default values
+        --> [float]:  copy all elements. Checks length.
+        """
+        if force_cmd is None:
+            self._data.force_command =  [self.default_force_command]*self.n_dim_cart
+        elif len(force_cmd) == self.n_dim_cart:
+            self._data.force_command = deepcopy(force_cmd)
+        else:
+            rospy.logerr('The number of force_command elements must be %d',
+                         self.n_dim_cart)
+
+    def set_interaction_frame(self, interaction_frame = None):
+        """
+        @param interaction_frame:
+        --> None:  populate with vector of default values
+        --> Pose: copy all the elements
+        """
+        if interaction_frame is None:
+            self._data.interaction_frame = Pose(
+                position=Point(x=0.0,y=0.0,z=0.0),
+                orientation=Quaternion(x=0.0,y=0.0,z=0.0,w=1.0))
+        elif isinstance(interaction_frame, Pose):
+            self._data.interaction_frame = interaction_frame
+        else:
+            rospy.logerr('interaction_frame is not of type geometry_msgs.Pose')
+
+    def set_endpoint_name(self, endpoint_name = None):
+        """
+        @param endpoint_name:
+        --> None:  set it to a default name (right hand)
+        --> string: copy element.
+        """
+        if endpoint_name is None:
```

Comment:
`self._data.endpoint_name = self.default_endpoint_name if endpoint_name is None else endpoint_name`
rather than add another function call to the stack.